pkg/controller/storagecluster/external_resources.go

```diff
@@ -182,3 +294,28 @@ func (r *ReconcileStorageCluster) createExternalStorageClusterSecret(sec *corev1
 	}
 	return nil
 }
+
+// To check if endpoint is a VALID ip and is REACHABLE or not
+func validateMonitoringEndpoint(monitoringIP string, monitoringPort string, reqLogger logr.Logger) error {
+	_, err := net.LookupIP(monitoringIP)
+	if err != nil {
+		reqLogger.Error(err, "Monitoring endpoint is not a valid IPv4 Ip.")
+		return err
+	}
+	endpoint := monitoringIP + ":" + monitoringPort
```

Comment:
Since we are in golang and already importing `net` package, use `net.JoinHostPort()` function (which will take care of IPV6 as well in the future)
```suggestion
	endpoint := net.JoinHostPort(monitoringIP, monitoringPort)
```
PS: again recommending that we use `MonitoringEndpoint` as single arg with `<IP>:<port>` value...

Reply:
@aruniiird Its how the rook script provides it, in separate arguments. If we need to change we will have to make a change in rook first.